### PR TITLE
Add a unit test for the `showRawYaml` option

### PR DESCRIPTION
**Pull request by Vlatombe**
Amends #459

### CODE REVIEW

**Comment by jglick** [comment]
```
java.util.ConcurrentModificationException
	at java.util.ArrayList$Itr.checkForComodification(ArrayList.java:909)
	at java.util.ArrayList$Itr.next(ArrayList.java:859)
	at org.csanchez.jenkins.plugins.kubernetes.pipeline.KubernetesPipelineTest.runInPod(KubernetesPipelineTest.java:106)
```

presumably a flake?

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java [hunk 1]
```diff
@@ -752,9 +752,8 @@ public void setShowRawYaml(boolean showRawYaml) {
     }
 
     private String getContainersDescriptionForLogging() {
-        List<ContainerTemplate> containers = getContainers();
         StringBuilder sb = new StringBuilder();
-        for (ContainerTemplate ct : containers) {
+        for (ContainerTemplate ct : getContainers()) {
```
Comment (jglick) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java]: Is this somehow related?

File: src/test/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplateTest.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    @Test
+    public void descriptionHonorsShowRawYaml() {
+        PodTemplate pt = new PodTemplate();
+        ContainerTemplate ct1 = new ContainerTemplate("foo", "image");
+        pt.setContainers(Collections.singletonList(ct1));
+        String yaml = new StringBuilder()
+                .append("metadata:").append("\n")
+                .append("  annotations:").append("\n")
+                .append("    foo: bar").append("\n")
+                .toString();
```
Comment (jglick) [inline src/test/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplateTest.java]: More readable as a simple series of strings concatenated with `+`.

**Comment by Vlatombe** [comment]
Yes, see https://github.com/jenkinsci/jenkins-test-harness/pull/144
